frontend/src/app/firewood/forests/buy-firewood-permit/buy-firewood-permit.component.ts

```diff
@@ -0,0 +1,263 @@
+    private router: Router,
+    private location: Location,
+    private titleService: Title,
+    public formBuilder: FormBuilder,
+    // public markdownService: NgxMdService,
+    public firewoodInfoService: FirewoodInfoService,
+    public applicationService: FirewoodApplicationService,
+    public applicationFieldsService: ApplicationFieldsService,
+    public util: UtilService,
+    private winRef: WindowRef,
+    private meta: Meta
+  ) {
+    this.meta.addTag({
+      name: 'description',
+      content: `Purchase a Firewood permit with the United States Forest Service on your National Forest.`
+    });
+    this.applicationForm = new FormGroup({
+      acceptPII: new FormControl(),
+      numberOfCords: new FormControl(),
+      firstName: new FormControl(),
+      lastName: new FormControl()
+    });
+  }
+
+  /**
+   * Update total cost when quantity changes
+   */
+  quantityChange(value) {
+    this.applicationForm.get('numberOfCords').setValidators([
+      Validators.required,
+      lessThanOrEqualValidator(4, 1)
+    ]);
+    if (!this.applicationForm.get('numberOfCords').errors) {
+      this.updateTotalCost();
+    } else {
+      this.applicationForm.get('totalCost').setValue(0);
+    }
+  }
+
+  /**
+   * @returns application form
+   */
+  getApplicationForm(formBuilder, maxCords) {
+    return formBuilder.group({
+      acceptPII: [false, Validators.required],
+      forestId: ['', [Validators.required]],
+      forestAbbr: [''],
+      // orgStructureCode: ['', [Validators.required]],
+      firstName: ['', [Validators.required, alphanumericValidator(), Validators.maxLength(36)]],
+      lastName: ['', [Validators.required, alphanumericValidator(), Validators.maxLength(60)]],
+      emailAddress: ['', [Validators.required, Validators.email, alphanumericValidator(), Validators.pattern('^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,4}$'), Validators.maxLength(255)]],
+      emailAddressConfirmation: [
+        '', [Validators.required, Validators.email, alphanumericValidator(), Validators.pattern(
+          '^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,4}$'
+        ), Validators.maxLength(255)]
+      ],
+      numberOfCords: ['', [Validators.required, Validators.min(1), Validators.max(maxCords)]],
+      // quantity: ['', [Validators.required, Validators.min(1), Validators.max(maxCords)]],
+      totalCost: [0, [Validators.required, currencyValidator()]]
+    },
+    {validator: emailConfirmationValidator('emailAddress', 'emailAddressConfirmation')});
+  }
+
+  /**
+   * Get application form and set default values
+   */
+  createForm(forest, formBuilder) {
+    this.applicationForm = this.getApplicationForm(formBuilder, 4);
+    this.applicationForm.get('acceptPII').setValue(false);
+    this.applicationForm.get('forestId').setValue(forest.id);
+    this.applicationForm.get('forestAbbr').setValue(forest.forestAbbr);
+    // this.applicationForm.get('orgStructureCode').setValue(forest.orgStructureCode);
```

Comment:
Are we using `orgStructureCode`? We should either reimplement this line or delete it.

Reply:
agreed removed